addons/survey/models/survey_survey.py

```diff
@@ -207,9 +209,9 @@ def _compute_page_and_question_ids(self):
     @api.depends('question_and_page_ids.is_conditional', 'users_login_required', 'access_mode')
     def _compute_is_attempts_limited(self):
         for survey in self:
-            if any(question.is_conditional for question in survey.question_and_page_ids)\
-                    or (survey.access_mode == 'public' and not survey.users_login_required)\
-                    or survey.is_attempts_limited is None:
+            if not survey.is_attempts_limited or \
+               any(question.is_conditional for question in survey.question_and_page_ids) or \
```

Comment:
We could put the condition about question_and_page_ids as third condition to use first fields defined on survey and hop we don't have to fetch sub models.

Reply:
and CRAC, blazing fast performances.